nixos/modules/misc/documentation.nix

```diff
@@ -187,7 +197,31 @@ in
       environment.systemPackages = [ pkgs.man-db ];
       environment.pathsToLink = [ "/share/man" ];
       environment.extraOutputsToInstall = [ "man" ] ++ optional cfg.dev.enable "devman";
-      environment.etc."man.conf".source = "${pkgs.man-db}/etc/man_db.conf";
+      environment.etc."man_db.conf".text =
+        let
+          getMan = pkg: if builtins.hasAttr "man" pkg then pkg.man else pkg;
+          manualPages = pkgs.symlinkJoin {
+            name = "man-paths";
+            paths = map getMan config.environment.systemPackages;
+          };
+          manualCache = pkgs.runCommandLocal "man-cache" { }
+          ''
+            echo "MANDB_MAP ${manualPages}/share/man $out" > man.conf
+            ${pkgs.man-db}/bin/mandb -C man.conf -psc
+          '';
+        in
+        ''
+          # Manual pages paths for NixOS
+          MANPATH_MAP /run/current-system/sw/bin /run/current-system/sw/share/man
+          MANPATH_MAP /run/wrappers/bin          /run/current-system/sw/share/man
+
+          ${optionalString cfg.man.generateCaches ''
+          # Generated manual pages cache for NixOS (immutable)
+          MANDB_MAP /run/current-system/sw/share/man ${manualCache}
```

Comment:
I don't think we need a man-db patch if we hardcode the cache path to e.g. `/etc/mandb`

Reply:
I'm not sure I understand. Are you proposing to symlink `manualCache` to  `/etc/mandb` or that the patch to check `/etc/man_db.conf` is not needed?

Reply:
We can compile it with a hardcoded config (which is supported without a patch) that adds a `MANDB_MAP` to `/etc/mandb`. Then we set `environment.etc.mandb.source = manualCache`

Reply:
Well, the patch is not great but it makes possible to configure man-db at "runtime", which may be useful besides adding the cache. Also `/etc` dosn't feel like the right place for a binary database.

Reply:
IMHO it would be better to remove `"--with-config-file=${placeholder "out"}/etc/man_db.conf"` from the man-db expression and put `man_db.conf` into `/etc` with `environment.etc`.

Reply:
@lheckemann That would be the best option but what about non-NixOS users?
If /etc/man_db.conf doesn't exist I don't think mandb programs will work at all (see https://github.com/NixOS/nixpkgs/issues/12097#issuecomment-168962194).
On debian there is no `man_db.conf`: the file is called `/etc/manpath.config`.

Reply:
Aah right, makes sense. Thanks for clearing that up :)